src/blockdata/script.rs

```diff
@@ -729,114 +729,94 @@ pub enum Instruction<'a> {
 
 /// Iterator over a script returning parsed opcodes.
 pub struct Instructions<'a> {
-    data: &'a [u8],
+    data: ::core::slice::Iter<'a, u8>,
     enforce_minimal: bool,
 }
 
+impl<'a> Instructions<'a> {
+    /// Set the iterator to end so that it won't iterate any longer
+    fn kill(&mut self) {
+        let len = self.data.len();
+        self.data.nth(len.max(1) - 1);
```

Comment:
Perhaps its easier to read like this because it saves having to think 'is it ok to call `self.data.nth(0)`, to kill an empty iterator'?
```rust
        let len = self.data.len();
        if len > 0 {
            self.data.nth(len - 1);
        }
```

Reply:
Yep, it was my first thought.